server/apps/publish/archive_publish.py

```diff
@@ -385,6 +377,80 @@ def publish(self, doc, updates, target_media_type=None):
 
         return queued
 
+    def sending_to_digital_subscribers(self, doc):
+        subscribers, subscribers_yet_to_receive = self.get_subscribers(doc)
+        filtered_subscribers = self.filter_subscribers(doc, subscribers, DIGITAL)
+        return len(filtered_subscribers) > 0
+
+    def get_subscribers(self, doc):
+        subscribers, subscribers_yet_to_receive = [], []
```

Comment:
Instead of multiple if conditions, it would be better to have this method in base class and override the same in sub-classes.